File: src/main/javascript/deploy_many/mem.js
var container = require('vertx/container');
var console = require('vertx/console');

function deploy_it(count) {
  container.deployVerticle('child.js', function(deploy_id) {
    console.log("deployed " + count);
    undeploy_it(deploy_id, count);
  });
}

function undeploy_it(deploy_id, count) {
  container.undeployVerticle(deploy_id, function() {
    count++;
    if (count < 10) {
      deploy_it(count);
    } else {
      console.log("done!");
    }
  });
}

deploy_it(0) ;

File: src/raw/ruby/deploy_many/mem.rb
require "vertx"

def deploy_it(count)
  Vertx::deploy_verticle("child.rb") do |err, deploy_id|
    if err
      puts "Failed to deploy #{err}"
    else
      puts "deployed #{count}"
      undeploy_it(deploy_id, count)
    end
  end
end

def undeploy_it(deploy_id, count)
  #puts "undeploying #{deploy_id}"
  Vertx::undeploy_verticle(deploy_id) do
    # puts "undeployed"
    count = count + 1
    if count < 10000
      deploy_it(count)
    else
      puts "done!"
    end
  end
end

deploy_it(0)
